### PR TITLE
SC | Switch to use platform profile

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > Switch Supplemental Claim contact info code to use the platform profile contact info definition. It's already being used by HLR & NOD
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
  > Switch to use platform defined contact info summary page and edit pages
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits decision reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#62612](https://github.com/department-of-veterans-affairs/va.gov-team/issues/62612)

## Testing done

- _Describe what the old behavior was prior to the change_
  > Removed duplicate tests
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
  > Platform contact info tests cover these changes
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

N/A

## What areas of the site does it impact?

Supplemental Claim

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by anniebtran — COMMENTED** [review]
generally looks good to me, but gonna defer approval to @rocio-desantiago since she probably has more context

File: src/applications/appeals/995/containers/App.jsx [hunk 1]
```diff
@@ -115,13 +94,6 @@ export const App = ({
             // resetStoredSubTask();
             setFormData({
               ...formData,
-              veteran: {
```
Comment (rocio-desantiago) [inline src/applications/appeals/995/containers/App.jsx:118]: so veteran data is included in platform profile?

File: src/applications/appeals/995/pages/contactInformation.js [hunk 2]
```diff
@@ -1,53 +1,27 @@
...
-          email: {
-            type: 'string',
-            format: 'email',
-            minLength: 6,
-            maxLength: 255,
-          },
-        },
-      },
+const allContacts = ['mobilePhone', 'homePhone', 'address', 'email'];
+const homelessContacts = ['mobilePhone', 'homePhone', 'email'];
```
Comment (rocio-desantiago) [inline src/applications/appeals/995/pages/contactInformation.js:8]: Is 995 the only form that uses this?

File: src/applications/appeals/995/tests/containers/App.unit.spec.jsx [hunk 3]
```diff
@@ -1,61 +1,18 @@
 import React from 'react';
-import moment from 'moment';
 import { render } from '@testing-library/react';
 import { Provider } from 'react-redux';
 import configureStore from 'redux-mock-store';
 import thunk from 'redux-thunk';
 import { expect } from 'chai';
 import sinon from 'sinon';
 
-import { VA_FORM_IDS } from 'platform/forms/constants';
```
Comment (rocio-desantiago) [inline src/applications/appeals/995/tests/containers/App.unit.spec.jsx:10]: are VA_FORM_IDS covered in the platform profile now?

**Review by rocio-desantiago — APPROVED** [review]
Just some questions

File: src/applications/appeals/995/containers/App.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/995/containers/App.jsx:118]: The `profileContactInfo` definition includes a [`wrapperKey` setting](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/definitions/profileContactInfo.js#L67) that defaults to `veteran`. What that means is the platform code builds the form data object using the provided wrapper key and adds all the contact info data inside, e.g.

```js
// form data
veteran: {
  mailingAddress: { /* ... */ },
  mobilePhone: { /* ... */ },
  homePhone: { /* ... */ },
  email: '...',
}
```

File: src/applications/appeals/995/pages/contactInformation.js (hunk 2 above)
Comment (Mottie) [reply src/applications/appeals/995/pages/contactInformation.js:8]: 995 is the only app we have that shows both home & mobile phone. All of our apps make the address (mailing address) not required if the Veteran indicates that they are homeless.

**Pull request merged**